bitnami/mariadb/templates/secrets.yaml

```diff
@@ -9,21 +9,25 @@ metadata:
     chart: "{{ template "mariadb.chart" . }}"
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
+  {{- if .Values.secret.annotations }}
+  annotations: {{- include "mariadb.tplValue" (dict "value" .Values.secret.annotations "context" $) | nindent 4 }}
+  {{- end }}
 type: Opaque
+{{- if .Values.secret.requirePasswords }}
 data:
   {{- if not (empty .Values.rootUser.password) }}
   mariadb-root-password: "{{ .Values.rootUser.password | b64enc }}"
   {{- else if (not .Values.rootUser.forcePassword) }}
   mariadb-root-password: "{{ randAlphaNum 10 | b64enc }}"
-  {{ else }}
+  {{ else if (.Values.rootUser.requirePassword) }}
```

Comment:
if you are going to add this new condition here, please, make this helper compatible with.

https://github.com/bitnami/charts/blob/377c51be686f0d5afa20e18f2f5935c93518ad38/bitnami/common/templates/_validations.tpl#L67

I think it would need to add just and `if` for each of them

```
{{- if (.Values.rootUser.requirePassword) -}}
{{- end -}}
```